client2/arq.go

```diff
@@ -0,0 +1,241 @@
+// SPDX-FileCopyrightText: © 2023 David Stainton
+// SPDX-License-Identifier: AGPL-3.0-only
+package client2
+
+import (
+	"errors"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/charmbracelet/log"
+
+	"github.com/katzenpost/katzenpost/core/crypto/rand"
+	sConstants "github.com/katzenpost/katzenpost/core/sphinx/constants"
+)
+
+const (
+	// MessageIDLength is the length of a message ID in bytes.
+	MessageIDLength = 16
+
+	// RoundTripTimeSlop is the slop added to the expected packet
+	// round trip timeout threshold.
+	RoundTripTimeSlop = 3 * time.Second
+)
+
+type SphinxComposerSender interface {
+	ComposeSphinxPacket(request *Request) ([]byte, []byte, time.Duration, error)
+	SendSphinxPacket(pkt []byte) error
+}
+
+// ARQMessage is used by ARQ.
+type ARQMessage struct {
+
+	// AppID identifies the application sending/receiving the message/reply.
+	AppID uint64
+
+	// ID is the unique message identifier
+	ID *[MessageIDLength]byte
```

Comment:
```suggestion
	MessageID *[MessageIDLength]byte
```
it's governed by `MessageID` length, and then we are not left at callsites wondering what kind of `id` something is

Reply:
it's the message id!

Reply:
fixed